### DISCUSSION TITLE
Bare metal IPI

### DISCUSSION THREAD

**Discussion by tcmal** [okd-project/okd#1224] [Q&A] [answered]
Hi all,

I'm trying to deploy an OKD cluster on bare metal but I'm a little unclear if IPI is supported yet? #197 seems like the right issue and is closed, but in practice I'm not able to do it because of the following error:

```
ERROR                                              
ERROR Error: Failed to query available provider packages 
ERROR                                              
ERROR Could not retrieve the list of available versions for provider 
ERROR openshift/local/libvirt: provider openshift/local/libvirt was not found in any 
ERROR of the search locations                      
ERROR                                              
ERROR   - /tmp/openshift-install-bootstrap-2428577597/plugins 
ERROR                                              
ERROR
```

Sure enough, the big list of symlinks that gets made right before this doesn't mention ironic. Is this known, and if so is there another tracking issue about it?

**Comment by vrutkovs** [comment, marked as answer]
libvirt should be present in openshift-baremetal-installer, but beware https://github.com/openshift/okd/discussions/1200 (should be fixed in next stable)

**Reply by tcmal** [reply]
Ah, perfect. Thank you!

Out of interest, is there a reason that baremetal installation is a seperate binary?

**Reply by vrutkovs** [reply]
libvirt requires linking to particular libvirt.so which won't work on Mac / some Linux distros. Previously we had this included in the main binary and a lot of people were unhappy about it
